lib/sycamore/sycamore/transforms/similarity.py

```diff
@@ -30,9 +30,7 @@ class SimilarityScorer(ABC):
         ignore_doc_structure: Ignore Document model (Document->Elements) in a DocSet
     """
 
-    def __init__(
-        self, ignore_element_sources: Optional[list[DocumentSource]] = None, ignore_doc_structure: bool = False
-    ):
+    def __init__(self, ignore_element_sources: Optional[list[str]] = None, ignore_doc_structure: bool = False):
```

Comment:
I think you should still be able to use an enum, but if not, can you use `typing.Literal` here

from typing import Literal
ignore_element_sources: Optional[List[Literal[DocumentSource.UNKNOWN, ...]]]

Reply:
Does this mean listing every literal supported? It's really just a string filter so I don't think its that useful here